charts/metering-ansible-operator/templates/crds/reportquery.crd.yaml

```diff
@@ -23,3 +23,103 @@ spec:
+
+  validation:
+  openAPIV3Schema:
+    required: ["spec"]
+    type: object
+    properties:
+      spec:
+        type: object
+        required: ["columns", "query"]
+        properties:
+          columns:
+            type: array
+            required: ["names"]
+            items:
+              type: object
+              required: ["type"]
+              properties:
+                name:
+                  type: string
+                  minLength: 1
+                type:
+                  type: string
+                  enum:
+                    - BOOLEAN
+                    - TINYINT
+                    - SMALLINT
+                    - INTEGER
+                    - BIGINT
+                    - REAL
+                    - DOUBLE
+                    - DECIMAL
+                    - VARCHAR
+                    - CHAR
+                    - VARBINARY
+                    - JSON
+                    - DATE
+                    - TIME
+                    - TIMESTAMP
+                    - ARRAY
+                    - MAP
+                    - MAP<VARCHAR, VARCHAR>
+                    - MAP<VARCHAR, INT>
+                    - MAP<INT, INT>
+                    - MAP<INT, VARCHAR>
+                    - ROW
+                    - IPADDRESS
+                    - UUID
+                    - HYPERLOGLOG
+                    - P4HYPERLOGLOG
+                    - QDIGEST
+                unit:
+                  type: string
+                  enum:
+                    - DATE
+                    - KUBERNETES_POD
+                    - KUBERNETES_PERSISTENTVOLUMECLAIM
+                    - KUBERNETES_PERSISTENTVOLUME
+                    - KUBERNETES_STORAGECLASS
+                    - KUBERNETES_NAMESPACE
+                    - BYTES
+                    - BYTE_SECONDS
+                    - TIME
+                    - CPU_CORE_SECONDS
+                    - CPU_CORES
+                    - MEMORY_BYTES
+                    - MEMORY_BYTES_SECONDS
+                    - SECONDS
+                tableHidden:
+                  type: boolean
+          inputs:
+            type: array
+            required: ["name"]
+            items:
+              type: object
+            properties:
+              name:
+                type: string
+                minLength: 1
+                type: boolean
+              type:
+                type: string
+                enum:
+                  - STRING
+                  - INTEGER
+                  - TIME
+                  - REPORTDATASOURCE
+                  - REPORTQUERY
+                  - REPORT
+              default:
+                type: string
+                oneOf:
+                  - type: string
+                  - type: boolean
+                  - type: number
+                  - type: null
+                  - type: array
+          query:
+            type: string
+            minLength: 1
+            pattern: ".*SELECT.*"
```

Comment:
I'd put this section at the top of the spec `properties`

Reply:
SELECT needs to be case-insensitive.